### PR TITLE
Add OAuth2AuthCredentials class and tests

### LINKED ISSUE #1028: We should add tests for AuthCredentials
Not all `AuthCredentials` subclasses are platform-dependent. In our unit tests we should cover those credentials whose creation is platform-independent.

### LINKED ISSUE #1029: Provide means to create AuthCredentials given an OAuth2 access token

### PR DESCRIPTION
**Pull request by mziccard**
This PR adds the `OAuth2AuthCredentials` class to fix #1029. Unit tests are added to fix #1028 

/cc @lesv

### CODE REVIEW

File: gcloud-java-core/src/test/java/com/google/cloud/AuthCredentialsTest.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import java.security.Signature;
+import java.security.SignatureException;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.Date;
+
+public class AuthCredentialsTest {
+
+  private static final String ACCESS_TOKEN = "accessToken";
+  private static final Date EXPIRATION_DATE = new Date();
```
Comment (lesv) [inline gcloud-java-core/src/test/java/com/google/cloud/AuthCredentialsTest.java]: Should this be a tiny bit in the future?

File: gcloud-java-core/src/test/java/com/google/cloud/SerializationTest.java [hunk 2]
```diff
@@ -94,7 +95,8 @@ public Builder toBuilder() {
   protected Restorable<?>[] restorableObjects() {
     try {
       return new Restorable<?>[]{AuthCredentials.createForAppEngine(), AuthCredentials.noAuth(),
-          AuthCredentials.createForJson(new ByteArrayInputStream(JSON_KEY.getBytes()))};
+          AuthCredentials.createForJson(new ByteArrayInputStream(JSON_KEY.getBytes())),
+          AuthCredentials.createFor("accessToken", new Date())};
```
Comment (lesv) [inline gcloud-java-core/src/test/java/com/google/cloud/SerializationTest.java]: Should the date be slightly in the future?

**Comment by lesv** [comment]
LGTM - You can ignore my ask about Date().  It would only be appropriate if somewhere you tested to see if a token / credential was still valid.  In looking through more of the code, I see that refresh is specifically disabled for some token kinds.  Somewhere there should be token expiration validation / refresh that happens, but not in any of your tests, nor the paths I traced through.

**Comment by mziccard** [comment]
@lesv Thanks for the pass.

Refresh is handled by `google-auth-library-java`'s classes. It happens when [`getRequestMetadata(Uri)`](https://github.com/google/google-auth-library-java/blob/master/oauth2_http/java/com/google/auth/oauth2/OAuth2Credentials.java#L70) is called on a credentials object with expired token.

**Comment by lesv** [comment]
Do you prefer that I merge your code on LGTM, or will you do it?  Thanks.

**Comment by mziccard** [comment]
I will take care of merging, thanks.

**Pull request merged**